modules/k8s-operator-crd-support/main.tf

```diff
@@ -109,6 +116,43 @@ module "k8sop_config" {
   kubectl_destroy_command = "kubectl delete -f - <<EOF\n${data.template_file.k8sop_config.rendered}EOF"
 }
 
+
+data "template_file" "rootsync_config" {
+
+  template = file(var.rootsync_cr_template_path)
+  vars = {
+    sync_repo          = var.sync_repo
+    sync_branch_node   = local.sync_branch_node
+    sync_revision_node = local.sync_revision_node
+    policy_dir_node    = local.policy_dir_node
+    secret_ref_node    = local.secret_ref_node
+    secret_type        = var.create_ssh_key ? "ssh" : var.secret_type
+    source_format_node = local.source_format_node
+  }
+}
+
+resource "time_sleep" "wait_for_configsync_api" {
```

Comment:
rather than sleep for 30, would it better to check if pods are up?
```
k -n config-management-system wait --for=condition=Ready pod --all
```

Reply:
Hi @bharathkkb, thanks for your review and sorry for my late reply.

I agree, checking the pods for readiness is a much better approach.
I added your suggestion to use kubectl wait. Unfortunately, in my first test the check failed because the config-management-system namespace wasn't created yet.
So, I added a script which checks the namespace to be created before checking the pods.

Reply:
Thanks @unlimitedbits  lgtm